File: CLASS2/18111.py
import sys
input = sys.stdin.readline
from collections import Counter

dx = [-1, 0, 1, 0]
dy = [0, -1, 0, 1]

n, m, b = map(int, input().split())
board = []
answer = (10**8, 0)
for _ in range(n):
    line = list(map(int, input().split()))
    board += line

total_block = sum(board)
board = Counter(board)

for i in range(257):
    time = 0
    if (n*m*i) - total_block > b:
        continue
    for key, val in board.items():
        if i < key:
            time += (key - i) * 2 * val
        elif i > key:
            time += (i - key) * val

    if answer[0] > time:
        answer = (time, i)
    elif answer[0] == time and answer[1] < i:
        answer = (time, i)

print(answer[0], answer[1])

File: solved.ac/CLASS2_JS/18111.js
const fs = require("fs");
const inputPath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const stdin = fs
  .readFileSync(inputPath)
  .toString()
  .split("\n")
  .map((line) => line.trim());

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const solution = () => {
  const [n, m, b] = input().split(" ").map(Number);
  const board = Array.from({ length: n }, () => input().split(" ").map(Number));
  let ansTime = Infinity;
  let ansHeight = 0;

  for (let z = 0; z <= 256; z++) {
    let total = b;
    let time = 0;
    for (let y = 0; y < n; y++) {
      for (let x = 0; x < m; x++) {
        if (board[y][x] >= z) {
          const temp = board[y][x] - z;
          total += temp;
          time += temp * 2;
        } else {
          const temp = z - board[y][x];
          total -= temp;
          time += temp * 1;
        }
      }
    }
    if (total >= 0 && ansTime >= time) {
      ansTime = time;
      ansHeight = z;
    }
  }
  console.log(ansTime, ansHeight);
};

solution();
